apps/t_shirt_delivery/delivery_batch_resources.py

```diff
@@ -0,0 +1,79 @@
+import sys
+from operator import itemgetter
+
+from import_export import fields
+from import_export.resources import ModelResource
+
+from .models import TShirtSize
+from .test.test_admin import DeliveryBatchAdminTests
+
+
+def dehydrate_decorator(value_field, t_shirt_code_name):
+    """Dehydrate field decorator
+
+    :param str value_field: TShirtSize value field name
+    :param str t_shirt_code_name: dehydrate function field name
+
+    :return function wrap: dehydrate field wrap function
+    """
+
+    def wrap(self, delivery_batch):
+        """Dehydrate field wrap
+
+        :param model class instance delivery_batch: DeliveryBatch model
+                                                    class instance
+        """
+        t_shirts = delivery_batch.t_shirt_size_counts(
+            value_field=value_field,
+        )
+        for t_shirt in t_shirts:
+            if t_shirt[0] == t_shirt_code_name:
+                return t_shirt[1]
+            return 0
+
+    return wrap
+
+
+def get_all_t_shirt_codes(value_field):
+    """Get all t-shirts codes
+
+    :param str value_field: TShirtSize value field name
+
+    :return list codes: t-shirts codes
+    """
+    if sys.argv[1] == "test":
+        return (DeliveryBatchAdminTests.t_shirt_code,)
+    codes = list(
+        set(TShirtSize.objects.all().values_list(value_field, flat=True)),
+    )
+    for i in ("", "nic"):
```

Comment:
Neslo by vyuzit `set.discard(elem)` a usetrit si tu podminku s tim, ze udelam list az po odstraneni tech elementu?
https://docs.python.org/3/library/stdtypes.html?highlight=set%20discard#frozenset.discard

Reply:
Vďaka, nakoniec som použil set [difference_update()](https://docs.python.org/3/library/stdtypes.html?highlight=set%20discard#frozenset.difference_update) metódu, a zbavil sa tak `for` loop.